sys-libs/stfl/stfl-0.24.recipe

```diff
@@ -0,0 +1,80 @@
+SUMMARY="A curses-based widget library"
+DESCRIPTION="STFL is a library which implements a curses-based widget set for \
+text terminals. The STFL API can be used from C, SPL, Python, Perl \
+and Ruby. Since the API is only 14 simple function calls big and there \
+are already generic SWIG bindings it is very easy to port STFL to \
+additional scripting languages.
+
+A special language (the Structured Terminal Forms Language) is used \
+to describe STFL GUIs. The language is designed to be easy and fast to \
+write so an application programmer does not need to spend ages \
+fiddling around with the GUI and can concentrate on the more \
+interesting programming tasks."
+HOMEPAGE="http://www.clifford.at/stfl/"
+COPYRIGHT="2006-2007  Clifford Wolf"
+LICENSE="GNU GPL v3"
+REVISION="1"
+SOURCE_URI="http://www.clifford.at/stfl/stfl-$portVersion.tar.gz"
+CHECKSUM_SHA256="d4a7aa181a475aaf8a8914a8ccb2a7ff28919d4c8c0f8a061e17a0c36869c090"
+SOURCE_DIR="stfl-$portVersion"
+PATCHES="stfl-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+PROVIDES="
+	stfl$secondaryArchSuffix = $portVersion compat >= 0.24
+	lib:libstfl$secondaryArchSuffix = $portVersion compat >= 0.24
+	lib:libstfl.so.$portVersion$secondaryArchSuffix = $portVersion compat >= 0.24
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libncurses$secondaryArchSuffix
+	lib:libiconv$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	stfl$secondaryArchSuffix = $portVersion compat >= 0.24
+	devel:libstfl$secondaryArchSuffix = $portVersion compat >= 0.24
+	devel:libstfl.so.0.24$secondaryArchSuffix = $portVersion compat >= 0.24
+	"
+REQUIRES_devel="
+	stfl$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libncurses$secondaryArchSuffix
+	devel:libiconv$secondaryArchSuffix
+	cmd:python
+	cmd:perl
+	devel:libruby$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc$secondaryArchSuffix
+	cmd:ld$secondaryArchSuffix
+	cmd:make
+	"
+# FOR BUILD() AND INSTALL()
+export libdir=$relativeLibDir
+export prefix=$prefix
+
+BUILD()
+{
+	make $jobArgs
+}
+
+INSTALL()
+{
+	make install
+	mkdir -p $includeDir
+	mv $prefix/include/* $includeDir
```

Comment:
It would be better to patch this directly in the makefile so it can be upstreamed